### PR TITLE
Fix Instance resource filters

### PR DESCRIPTION
**Pull request by ooga**

### CODE REVIEW

File: source/tyr/tyr/resources.py [hunk 1]
```diff
@@ -272,7 +272,11 @@ def get(self, id=None, name=None):
         parser.add_argument('is_free', type=inputs.boolean, required=False,
                 case_sensitive=False, help='boolean for returning only free or private instances')
         args = parser.parse_args()
-        if args['is_free'] != None:
+        if args['is_free'] is None:
+            del args['is_free']
```
Comment (xlqian) [inline source/tyr/tyr/resources.py]: Something may be missing for me, Why you need to `del` it?

Comment (xlqian) [inline source/tyr/tyr/resources.py]: Ok, I understand...

Comment (xlqian) [inline source/tyr/tyr/resources.py]: Suggestion:  

```
        #if args['is_free'] is None:
        #    del args['is_free']
        args.update({'id': id, 'name': name})
        if any((v is not None for i in args.values())):
```

**Comment by ooga** [comment]
@xlqian thanks for the suggestion

File: source/tyr/tyr/resources.py [hunk 2]
```diff
@@ -272,8 +272,10 @@ def get(self, id=None, name=None):
         parser.add_argument('is_free', type=inputs.boolean, required=False,
                 case_sensitive=False, help='boolean for returning only free or private instances')
         args = parser.parse_args()
-        if args['is_free'] != None:
-            return models.Instance.query.filter_by(**args).all()
+        args.update({'id': id, 'name': name})
+        if any(v is not None for v in args.values()):
+            filters = {k: v for k, v in args.items() if v is not None}
```
Comment (xlqian) [inline source/tyr/tyr/resources.py]: Sorry for bothering you again, do you really need this line? Will it still work if you remove this line?

**Comment by xlqian** [comment]
ok for me

**Comment by xlqian** [comment]
Jekins, test this please!

**Comment by xlqian** [comment]
Jenkins, test this please!

**Comment by xlqian** [comment]
Jenkins, test this please!

**Comment by antoine-de** [comment]
thx!

**Pull request merged**